File: frontend/src/types/scheduler.ts
/** Scheduler types for PRD v3.2 FR-SS002/SS004. */

export interface SchedulerStatus {
  current_profile: string
  override: string | null
  is_trading_day: boolean
  current_session: string
  is_holiday_period: boolean
  next_trading_day: string
}

export interface TaskConfig {
  name: string
  enabled: boolean
  description: string
}

export interface SchedulePlan {
  name: string
  label: string
  default_enabled: boolean
  tasks: TaskConfig[]
}

export interface SchedulePlansResult {
  plans: SchedulePlan[]
}

export interface CalendarDay {
  date: string
  is_trading_day: boolean
  is_weekend: boolean
  is_holiday: boolean
  day_of_week: number
}

export interface CalendarResult {
  days: CalendarDay[]
  today: string
  next_trading_day: string
}

export interface NotificationChannel {
  type: string
  enabled: boolean
  events: string[]
  webhook_url?: string
  bot_token?: string
  chat_id?: string
  secret?: string
  url?: string
  method?: string
  headers?: Record<string, string>
}

export interface SentinelConfig {
  data_sources: {
    global_markets?: { enabled: boolean; provider: string }
    news_platforms?: { enabled: boolean; platforms: string[]; refresh_interval: number }
    rss_feeds?: { enabled: boolean; feeds: { url: string; name: string; category: string }[]; max_age_days: number }
  }
  notifications: {
    channels: NotificationChannel[]
    event_types: string[]
  }
}

File: src/web/schemas/scheduler.py
"""Pydantic models for scheduler API endpoints.

Per PRD v3.2 FR-SS004.
"""

from __future__ import annotations

from pydantic import BaseModel


class SchedulerStatus(BaseModel):
    """Current scheduler status."""

    current_profile: str
    override: str | None
    is_trading_day: bool
    current_session: str
    is_holiday_period: bool
    next_trading_day: str


class TaskConfig(BaseModel):
    """Individual task configuration within a plan."""

    name: str
    enabled: bool
    description: str = ""


class SchedulePlan(BaseModel):
    """A schedule plan (e.g. trading_day, holiday)."""

    name: str
    label: str
    default_enabled: bool
    tasks: list[TaskConfig]


class SchedulePlansResult(BaseModel):
    """All schedule plans."""

    plans: list[SchedulePlan]


class CalendarDay(BaseModel):
    """Single day in the calendar view."""

    date: str
    is_trading_day: bool
    is_weekend: bool
    is_holiday: bool
    day_of_week: int


class CalendarResult(BaseModel):
    """30-day calendar view."""

    days: list[CalendarDay]
    today: str
    next_trading_day: str


class UpdatePlanRequest(BaseModel):
    """Request to update task configs within a plan."""

    tasks: dict[str, bool]


class SentinelConfig(BaseModel):
    """Sentinel configuration (data sources + notifications)."""

    data_sources: dict
    notifications: dict


class NotificationChannelConfig(BaseModel):
    """Configuration for a notification channel."""

    type: str
    enabled: bool
    events: list[str] = []
